homeassistant/components/__init__.py

```diff
@@ -143,9 +143,11 @@ def async_handle_turn_service(service):
         yield from asyncio.wait(tasks, loop=hass.loop)
 
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service,
+        state_to_set=STATE_OFF)
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service,
+        state_to_set=STATE_ON)
```

Comment:
Do you mean a dict for lookup like we have today in the state helper? That's not working well today for many reasons. One is that it imports from the components. Another is that it needs manual updating whenever we add a new service, so it's always behind, since it's not directly coupled to the registration of the services.

Maybe I'm still missing your point?